tls/s2n_signature_scheme.c

```diff
@@ -142,8 +142,35 @@ const struct s2n_signature_scheme s2n_rsa_pss_rsae_sha512 = {
         .signature_curve = NULL /* Elliptic Curve not needed for RSA */
 };
 
-/* Signature Scheme Preference List: TLS 1.2 and previous */
-/* This list MUST NOT contain any s2n_signature_scheme's with a non-null signature_curve defined. */
+/* All Supported SignatureSchemes (Both TLS 1.2 and 1.3) to send in the ClientHello to the Server. */
+/* No MD5 to avoid SLOTH Vulnerability */
+const struct s2n_signature_scheme* const s2n_supported_sig_scheme_pref_list[] = {
+        /* RSA PSS - Commented out until it is actually supported */
+        /* &s2n_rsa_pss_rsae_sha256, */
+        /* &s2n_rsa_pss_rsae_sha384, */
+        /* &s2n_rsa_pss_rsae_sha512, */
+
+        /* RSA PKCS1 */
+        &s2n_rsa_pkcs1_sha256,
+        &s2n_rsa_pkcs1_sha384,
+        &s2n_rsa_pkcs1_sha512,
+        &s2n_rsa_pkcs1_sha224,
+        &s2n_rsa_pkcs1_sha1,
```

Comment:
should the sha1 algorithms (s2n_rsa_pkcs1_sha1,  s2n_ecdsa_sha1 ) be placed at the bottom of the list? see "Legacy algorithms" under https://tools.ietf.org/html/rfc8446#page-42

Reply:
Fixed